lib/runners/processor/detekt.rb

```diff
@@ -0,0 +1,371 @@
+
+        check_runner_config(config) do |checked_config|
+          @detekt_config = checked_config
+
+          response = 
+            case
+            when gradle_config
+              run_gradle
+            when maven_config
+              run_maven
+            else
+              run_cli
+            end
+
+          if response.kind_of?(Results::Failure)
+            return response
+          end
+          
+          Results::Success.new(guid: guid, analyzer: analyzer).tap do |result|
+            response.each do |issue|
+              result.add_issue issue
+            end
+          end
+        end
+      end
+    end
+
+    def analyzer_version
+      unknown_version = "0.0.0"
+
+      @analyzer_version ||=
+        case
+        when gradle_config
+          extract_detekt_version_for_gradle || unknown_version
+        when maven_config
+          extract_detekt_version_for_maven || unknown_version
+        else
+          pom_dir = Pathname(ENV['RUNNER_USER_HOME']) / analyzer_name
+          extract_detekt_version_for_maven(pom_dir) || unknown_version
+        end
+    end
+
+    def detekt_config
+      @detekt_config
+    end
+
+    def cli_config
+      detekt_config[:cli]
+    end
+
+    def gradle_config
+      detekt_config[:gradle]
+    end
+
+    def maven_config
+      detekt_config[:maven]
+    end
+
+    def run_cli
+      report_id = "xml"
+      report_path = Tempfile.new("detekt-report")
+      
+      args = [
+        cli_baseline,
+        cli_config_files,
+        cli_config_resource,
+        cli_disable_default_rulesets,
+        cli_excludes,
+        cli_input,
+        cli_report(report_id, report_path.path)
+      ].flatten.compact
+
+      _stdout, stderr, status = capture3(analyzer_bin, *args)
+
+      # detekt has some exit codes.
+      # @see https://github.com/arturbosch/detekt/blob/1.6.0/docs/pages/gettingstarted/cli.md
+      case status.exitstatus
+      when 0, 2
+        construct_result(report_id, report_path)
+      when 3 # invalid configuration
+        Results::Failure.new(guid: guid, message: "Your detekt configuration is invalid", analyzer: analyzer)
+      else
+        trace_writer.error stderr.strip
+        raise stderr
+      end
+    end
+
+    def run_gradle
+      capture3("./gradlew", "--quiet", gradle_config[:task])
+      construct_result( 
+        gradle_config[:report_id],
+        gradle_config[:report_path]
+      )
+    end
+
+    def run_maven
+      mvn_options = %w[--quiet --batch-mode -Dmaven.test.skip=true -Dmaven.main.skip=true]
+      capture3("mvn", maven_config[:goal], *mvn_options)
+      construct_result(
+        maven_config[:report_id],
+        maven_config[:report_path]
+      )
+    end
+
+    def construct_result(report_id, report_path)
+      output_file_path = current_dir / report_path
+      if output_file_path.exist?
+        trace_writer.message "Reading output from #{report_path}..."
+        output = output_file_path.read
```

Comment:
[suggestion] Can you add `output` to trace for debugging?

```suggestion
        output = output_file_path.read
        trace_writer.message output
```